Update keep car awake using Sentry Mode via TeslaFi & Tessie API. Deprecate obsolete Tesla API codes.

THX723 opened the pull request:
Sentry Mode is now the only viable API method for keeping the car awake during archiving.
`awake_start` & `awake_stop` have been rewritten and any legacy/Tesla API related codes have been removed.
New parameter required: `SENTRY_CASE`
See updated description in `.conf` file for additional details.

Legacy method of keeping track of previous Sentry Mode state is no longer relevant nor necessary. The general mode of operation can be far simpler and broken into the following 3 scenarios:

Case 0: Those who prefer to have Sentry Mode ON anywhere.
No special actions is required from TeslaUSB. With Sentry Mode always ON, TeslaUSB will be consistently powered and able to complete the archive without interruptions.

Case 1: Those who prefer to have Sentry Mode ON anywhere, except Home.
This is the case for me. I don't use Sentry Mode when the car is safely parked inside an enclosed garage.
Sentry Mode must be set to ON in-car and DO NOT select 'Exclude Home' (counter intuitive).
No special actions for TeslaUSB before archiving, since Sentry will already have been ON when first arriving home.
Sentry OFF is executed when archive completes to allow the car to fall asleep.
Whenever the car leaves home, Sentry Mode will automatically default back to ON per the in-car setting.

Case 2: Those who prefer to have Sentry Mode OFF anywhere.
Sentry Mode must be set to OFF in-car. 
Right before archiving, TeslaUSB will execute a Sentry Mode ON command.
Sentry OFF is executed when archive completes to allow the car to fall asleep.
Whenever the car leaves home, Sentry Mode will continue to be OFF per the in-car setting.

THX723 committed:
Update awake_start for Sentry Mode via TeslaFi and Tessie. Deprecate obsolete Tesla API codes.

Reference revised .conf file for additional details.

THX723 committed:
Update awake_stop for Sentry Mode via TeslaFi and Tessie. Deprecate obsolete Tesla API codes.

Reference revised .conf file for more details.

THX723 committed:
Update teslausb_setup_variables.conf.sample for Sentry Mode via TeslaFi and Tessie.

New parameter required for TeslaFi and Tessie: SENTRY_CASE.
Read new description for mode of operation.

marcone commented:
Do you want to also remove tesla_api.py and the code that calls it in configure.sh?

THX723 commented:
> Do you want to also remove tesla_api.py and the code that calls it in configure.sh?

I wonder if that'll ever get updated to support the latest Tesla changes. Probably not. Let me dig into this a bit more.

THX723 committed:
Update configure.sh deprecate obsolete Tesla API codes.

No longer need to install TeslaPy, as Tesla deprecated legacy/unofficial API support.

THX723 commented:
Does not look like TeslaPy will ever get updated moving forward. I have now removed legacy Tesla API installs and checks in `configure.sh`
